### DISCUSSION TITLE
Datepicker: when rendered with ui-datepicker-touch-ui

### DISCUSSION THREAD

**Discussion by rho66** [primefaces/community#295] [PrimeFaces] [answered]
Under which circumstances is Datepicker rendered with `ui-datepicker-touch-ui`?

I have to dialogs using Dialog Framework. In one dialog datepicker is render without `ui-datepicker-touch-ui` and in the other one with `ui-datepicker-touch-ui`?

Problem is that `ui-datepicker-touch-ui` renders with
```
.ui-datepicker-touch-ui.ui-datepicker td a, 
.ui-datepicker-touch-ui.ui-datepicker td span,
.ui-datepicker-touch-ui.ui-datepicker th {
    padding: 2em 0;
}
```
from primefaces `components.css`.

Even using datapicker with attributes `touchable="false" touchUI="false"` datepicker is rendered with `ui-datepicker-touch-ui`!

All my other usages of datepicker don't have this "big" padding!

**Comment by melloware** [comment, marked as answer]
You didn't mention your version of PrimeFaces?

Also did you try...`autoDetectDisplay="false"`

```xml
<p:datePicker id="touch" value="#{calendarView.date4}" touchUI="false" showOtherMonths="true" autoDetectDisplay="false" />
```

**Comment by rho66** [comment]
I am using PF 12.0.4

**Comment by rho66** [comment]
`autoDetectDisplay="false"` solved my problem!

big thanks to @melloware
